sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosTracerTest.java

```diff
@@ -0,0 +1,245 @@
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        cosmosAsyncContainer.getScripts().readAllUserDefinedFunctions(new CosmosQueryRequestOptions()).byPage().single().block();
+        Mockito.verify(tracer, Mockito.times(7)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        cosmosAsyncContainer.getScripts().getUserDefinedFunction(cosmosUserDefinedFunctionProperties.getId()).delete().block();
+        Mockito.verify(tracer, Mockito.times(8)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        CosmosTriggerProperties cosmosTriggerProperties = getCosmosTriggerProperties();
+        CosmosTriggerProperties resultTrigger =
+            cosmosAsyncContainer.getScripts().createTrigger(cosmosTriggerProperties).block().getProperties();
+        Mockito.verify(tracer, Mockito.times(9)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        cosmosAsyncContainer.getScripts().getTrigger(cosmosTriggerProperties.getId()).read().block();
+        Mockito.verify(tracer, Mockito.times(10)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        cosmosAsyncContainer.getScripts().getTrigger(cosmosTriggerProperties.getId()).replace(resultTrigger).block();
+        Mockito.verify(tracer, Mockito.times(11)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        cosmosAsyncContainer.getScripts().readAllTriggers(new CosmosQueryRequestOptions()).byPage().single().block();
+        Mockito.verify(tracer, Mockito.times(12)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        cosmosAsyncContainer.getScripts().getTrigger(cosmosTriggerProperties.getId()).delete().block();
+        Mockito.verify(tracer, Mockito.times(13)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        CosmosStoredProcedureProperties procedureProperties = getCosmosStoredProcedureProperties();
+        CosmosStoredProcedureProperties resultSproc =
+            cosmosAsyncContainer.getScripts().createStoredProcedure(procedureProperties).block().getProperties();
+        Mockito.verify(tracer, Mockito.times(14)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        cosmosAsyncContainer.getScripts().getStoredProcedure(procedureProperties.getId()).read().block();
+        Mockito.verify(tracer, Mockito.times(15)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
```

Comment:
Could we update these tests to check for the `attributes` that are needed to be set for the particular request, rather than just checking for `anyString()`?

Reply:
This comment was also there in last pr which was reverted , and as I stated earlier we will do much more attributes check once we will have the diagnostic tracer support pr , which is next in line. And also as mentioned in the description of this PR that scope of this pr is basically re merging earlier PR without short circuiting. 
Can we please this take this in next PR unless this is blocker. Also I manually verified  the attributes in Jagger UI .

Reply:
> Also I manually verified the attributes in Jagger UI .

Suggesting the same attributes to be checked when making the tracer calls. Sill related to this PR work and not so concerned about the diagnostic information which should be future work PR items.

If you want to do it ina  follow up PR, please open an issue to track it. But it is still a part of this feature testing.

Reply:
Checking attributes will be extension of this test class. It will be non minimal work, create github issue assigned to me
https://github.com/Azure/azure-sdk-for-java/issues/12988

Reply:
Attributes checking is added in the test case